main/high/src/reaper.rs

```diff
@@ -441,17 +443,79 @@ impl Reaper {
         let registered_action = RegisteredAction::new(command_id);
         // Immediately register if active
         let mut session_status = reaper_main.session_status.borrow_mut();
-        let awake_state = match session_status.deref_mut() {
-            SessionStatus::Sleeping => return registered_action,
-            SessionStatus::Awake(s) => s,
-        };
-        let action_reg = register_action(
-            &mut medium,
-            command_id,
-            description.into_owned(),
-            default_key_binding,
+        if let SessionStatus::Awake(s) = session_status.deref_mut() {
+            // We don't need to store the custom action handle since it's managed differently
+            // than regular action handles
+        }
+        registered_action
+    }
+
+    /// Registers an action in a specific section (e.g. MIDI editor, media explorer, etc.)
+    /// 
+    /// # Arguments
+    /// * `command_name` - The unique command name for the action
+    /// * `description` - The description shown in the action list
+    /// * `section_id` - The section ID where the action should be registered (e.g. 32060 for MIDI editor)
+    /// * `operation` - The function to execute when the action is triggered
+    /// * `kind` - Whether the action is toggleable or not
+    /// 
+    /// # Returns
+    /// A `RegisteredAction` that can be used to unregister the action later
+    pub fn register_action_in_section(
```

Comment:
It looks like you are using the high-level API. But its usage is highly deprecated because it's very volatile. Not sure if you are aware of that?